controllers/vspherevm_controller.go

```diff
@@ -80,7 +86,34 @@ func AddVMControllerToManager(ctx *context.ControllerManagerContext, mgr manager
 			&source.Channel{Source: ctx.GetGenericEventChannelFor(controlledTypeGVK)},
 			&handler.EnqueueRequestForObject{},
 		).
-		Complete(vmReconciler{ControllerContext: controllerContext})
+		Build(r)
+
+	if err != nil {
+		return err
+	}
+
+	err = controller.Watch(
```

Comment:
nit, this should be `if err := controller.Watch` to descope the error.

Reply:
Can probably also just simplify to `return controller.Watch(...)`, since we are just returning the `err` in the non-nil case and nil otherwise.